packages/react-native-reanimated/src/hook/commonTypes.ts

```diff
@@ -93,9 +94,11 @@ export interface AnimatedStyleHandle<
 }
 
 export interface JestAnimatedStyleHandle<
-  Style extends DefaultStyle = DefaultStyle,
+  Style extends DefaultStyle | AnimatedProps = DefaultStyle,
 > extends AnimatedStyleHandle<Style> {
-  jestAnimatedStyle: MutableRefObject<AnimatedStyle<Style>>;
+  jestAnimatedValues:
+    | MutableRefObject<AnimatedStyle<Style>>
+    | MutableRefObject<AnimatedProps>;
```

Comment:
Shouldn't it be an intersection?

```suggestion
  jestAnimatedValues:
     MutableRefObject<AnimatedStyle<Style>>
    & MutableRefObject<AnimatedProps>;
```

Reply:
Yeah, changing this bit would require a refactor, let's leave it as is then.